### ISSUE TITLE
[2] GAIN: Missing Data Imputation using Generative Adversarial Nets

### ISSUE THREAD

**Issue by techzzt** [techzzt/pytorch-paper-implement#2]
> **GAIN: Missing Data Imputation using Generative Adversarial Nets**

Table 형식의 데이터의 결측 값을 imputation하기 위해 generative 기반의 모델을 사용한 방법론 

![image](https://user-images.githubusercontent.com/49440852/176133692-49b42a22-7d35-4611-8eaf-ec37a0c4267a.png)

Input에 대해 결측 값이 존재하는 구간에 대한 정보를 추가하기 위해 입력 데이터만을 사용하는 것이 아닌, hint matrix와 mask matrix를 함께 모델의 입력 값으로 사용 

Data matrix (원본 값), random matrix (결측 영역에 random하게 imputation한 matrix), mask matrix (결측 영역 정보 포함)을 generator의 입력 값으로 사용 -> 결측 구간에 추정된 값이 채워진 imputed matrix와 mask matrix에서 결측 영역의 정보 포함 여부를 저장한 hint mask를 discriminator의 입력 값으로 사용 -> 이후 discriminator의 결과 값으로 마스크 영역이 모두 추정된 estimated mask matrix를 출력해 원본 mask matrix와 loss (cross entropy loss)를 계산해가며 학습  

![image](https://user-images.githubusercontent.com/49440852/176133858-752589ad-7f42-4b81-ab32-90f4d3c99dcd.png)
